modules/cudacodec/src/video_decoder.hpp

```diff
@@ -70,6 +70,7 @@ class VideoDecoder
     void create(const FormatInfo& videoFormat);
     int reconfigure(const FormatInfo& videoFormat);
     void release();
+    bool inited() { AutoLock autoLock(mtx_); return decoder_; }
```

Comment:
`decoder_` is not protected by `mtx_`. Only format modification does. Also decoder_value may be modified in the middle  of `cuvidCreateDecoder` call internally, e.g. just allocated memory for the structure and the check fails.  IMHO the original version was better, even if it's not truly thread safe.

Reply:
What about placing the lock around the decoder creation as well so this version is thread safe?